### DISCUSSION TITLE
uninitialized constant SolidusPaypalBraintree::Gateway (NameError)

### DISCUSSION THREAD

**Discussion by czepesch** [solidusio/solidus#4746] [Troubleshooting] [answered]
Hello!
I'm getting `uninitialized constant SolidusPaypalBraintree::Gateway (NameError)` when trying to start the server after configuring SolidusPaypalBraintree gem exactly how README says.
What it can be?
Will appreciate any hints.

ruby 3.1.2
rails 7.0.4

**Comment by waiting-for-dev** [comment, marked as answer]
Since Rails 7 is no longer possible to reference reloadable constants in an initializer. Probably, you need to change to:

```ruby
Rails.application.config.to_prepare do
  Spree::Config.static_model_preferences.add(
     SolidusPaypalBraintree::Gateway,
    'braintree_credentials', {
       environment: Rails.env.production? ? 'production' : 'sandbox',
       merchant_id: ENV['BRAINTREE_MERCHANT_ID'],
       public_key: ENV['BRAINTREE_PUBLIC_KEY'],
       private_key: ENV['BRAINTREE_PRIVATE_KEY'],
       paypal_flow: 'vault', # 'checkout' is accepted too
       use_data_collector: true # Fingerprint the user's browser when using Paypal
     }
  )
end
```

I created a [PR to update the README](https://github.com/solidusio/solidus_paypal_braintree/pull/332).

Thanks for reporting.
